src/Analyzer/Utils.h

```diff
@@ -83,4 +85,9 @@ void replaceColumns(QueryTreeNodePtr & node,
     const QueryTreeNodePtr & table_expression_node,
     const std::unordered_map<std::string, QueryTreeNodePtr> & column_name_to_node);
 
+/** Resolve function node again using it's content.
+  * This function should be called when arguments or parameters are changed.
+  */
+void rerunFunctionResolve(FunctionNode * function_node, ContextPtr context);
```

Comment:
I suppose we do this in many places in optimization passes already, just wondering why we haven't had such function yet.

Reply:
Well, because it's hard to write a function with would support all the necessary cases. I'll leave this function in the code, but I won't use it in `IfConstantConditionPass`. There are quite a lot of cases when we can't resolve a node again.